pkg/api/datavolume/store.go

```diff
@@ -40,13 +43,30 @@ func (s *dvStore) canDelete(namespace, name string) error {
 		return fmt.Errorf("failed to get dv %s, %v", name, err)
 	}
 
-	if len(dv.OwnerReferences) != 0 {
-		for _, owner := range dv.OwnerReferences {
-			if owner.Kind == kv1alpha3.VirtualMachineGroupVersionKind.Kind {
-				return fmt.Errorf("can not delete the volume %s which is currently owned by the VM %s", name, owner.Name)
-			}
+	annotationSchemaOwners, err := ref.GetSchemaOwnersFromAnnotation(dv)
+	if err != nil {
+		return fmt.Errorf("failed to get schema owners from annotation: %w", err)
+	}
+
+	attachedList := annotationSchemaOwners.List(kv1alpha3.VirtualMachineGroupVersionKind.GroupKind())
+	if len(attachedList) != 0 {
+		return fmt.Errorf("can not delete the volume %s which is currently attached by these VMs: %s", name, strings.Join(attachedList, ","))
+	}
+
+	if len(dv.OwnerReferences) == 0 {
+		return nil
+	}
+
+	var ownerList []string
+	for _, owner := range dv.OwnerReferences {
+		if owner.Kind == kv1alpha3.VirtualMachineGroupVersionKind.Kind {
+			ownerList = append(ownerList, owner.Name)
```

Comment:
why don't we do return in here instead?

Reply:
I am not sure if a volume only can be mounted by a virtual machine.